src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -1360,4 +1363,39 @@ public boolean shouldTick() {
     public void setInvulnerable(boolean value) {
         this.invulnerable = value;
     }
+
+    @Inject(method = "setFire", at = @At(value = "HEAD"), cancellable = true)
```

Comment:
Need help, setting this to RETURN doesn't fire the event at all.

Reply:
At RETURN is `isBurning` probably already returning `true`.

You will need to implement the injection differently if you want to support the overridden fire seconds. A `ModifyVariable` should do the trick. Inject into the seconds argument and return the modified seconds from the event. `isBurning` will also work at this point.

Reply:
You can even get around by using a `@Redirect` with a field setter. This way you can cancel the return by sending the current fire seconds, or increase it based on whatever event is thrown.

Reply:
I was testing around and ended up with:
```
@Inject(method = "setFire",
            at = @At(value = "FIELD", target = "Lnet/minecraft/entity/Entity;fire:I", opcode = Opcodes.PUTFIELD),
            locals = LocalCapture.CAPTURE_FAILEXCEPTION,
            cancellable = true)
```
Everything is working, not sure what is better.